### PR TITLE
Automatically Generate 403 Responses

### PR DESCRIPTION
**Pull request by mureytasroc**

### CODE REVIEW

File: backend/PennCourses/docs_settings.py [hunk 1]
```diff
@@ -866,53 +866,86 @@ def _get_responses(self, path, method):
...
         ):
             if path in self.override_schema and method in self.override_schema[path].keys():
                 return {
+                    **{
+                        "200": {
+                            "description": "",
+                            **(
+                                {
+                                    "content": {
+                                        ct: {"schema": self.override_schema[path][method][200]}
```
Comment (davish) [inline backend/PennCourses/docs_settings.py:889]: I'm guilty of this too, but could you spell out what this variable is? one and two letter variable names can get confusing.

Comment (davish) [reply backend/PennCourses/docs_settings.py:889]: looks like it's `content_type`

File: backend/PennCourses/docs_settings.py [hunk 2]
```diff
@@ -866,53 +866,86 @@ def _get_responses(self, path, method):
...
+                        # TODO: put something meaningful into it
                         "description": "",
+                    }
+                },
+                **auth_response_dict,
+            }
+        if path in self.override_schema and method in self.override_schema[path].keys():
+            return {
+                **{
+                    str(k): {
```
Comment (davish) [inline backend/PennCourses/docs_settings.py:920]: same here. could you replace `k` and `v` with something a bit more specific? `k` seems to be the status code and `v` seems to be the message.

File: backend/PennCourses/docs_settings.py [hunk 3]
```diff
@@ -866,53 +866,86 @@ def _get_responses(self, path, method):
...
                                     for ct in self.response_media_types
                                 }
-                                if 200 in self.override_schema[path][method].keys()
+                                if k in self.override_schema[path][method].keys()
                                 else {
                                     ct: {"schema": response_schema}
                                     for ct in self.response_media_types
                                 }
                             }
+                            if "[SCHEMA]" in v
```
Comment (davish) [inline backend/PennCourses/docs_settings.py:934]: This should be documented -- ideally there'd be a better way to do this than a string keyword, but this is fine as long as it's documented (both at the top [ where i think it is ] and right here.

**Review by davish — COMMENTED** [review]
some small naming things, but after that looks good to go!!!

File: backend/PennCourses/docs_settings.py (hunk 1 above)
Comment (mureytasroc) [reply backend/PennCourses/docs_settings.py]: This naming convention was from the DRF AutoSchema code but you're definitely right that `content_type` is better.

File: backend/PennCourses/docs_settings.py (hunk 2 above)
Comment (mureytasroc) [reply backend/PennCourses/docs_settings.py]: Done.  Good point, will make sure to use more descriptive names in the future.

File: backend/PennCourses/docs_settings.py (hunk 3 above)
Comment (mureytasroc) [reply backend/PennCourses/docs_settings.py]: It is documented at the top, and I documented it here as well.

**Review by davish — APPROVED** [review]
Let's rock!!

**Pull request merged**